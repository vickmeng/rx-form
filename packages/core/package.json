{
  "name": "@yzw-rx-form/core",
  "version": "1.0.0",
  "description": "core",
  "module": "dist/index.js",
  "types": "dist/index.d.ts",
  "main": "dist/index.js",
  "private": false,
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "rollup -c",
    "watch": "rollup -c --watch",
    "test": "jest -c",
    "test:c": "pnpm run test --coverage"
  },
  "author": "laomeng",
  "license": "ISC",
  "dependencies": {
    "lodash": "^4.0.0"
  },
  "peerDependencies": {
    "rxjs": "*"
  },
  "devDependencies": {
    "rxjs": "*"
  }
}
