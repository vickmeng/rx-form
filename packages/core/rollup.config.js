import typescript from '@rollup/plugin-typescript';

export default {
  input: 'src/index.ts',
  output: {
    file: 'dist/index.js',
    format: 'es',
  },
  external: ['lodash', 'rxjs'],
  plugins: [
    typescript({
      tsconfig: './tsconfig.json',
    }),
  ],
};
