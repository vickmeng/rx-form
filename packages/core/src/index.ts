/**
 * Controls
 */

export { FieldControl } from './controls/fieldControl';
export { GroupControl } from './controls/groupControl';
export { ListControl } from './controls/listControl';
export { BaseControl } from './controls/baseControl';

export type {
  FormControlOptions,
  FormGroupOptions,
  FormListOptions,
  FormGroupControlsConfig,
  FormListControlsConfig,
  ValidatorFn,
  AbstractControl,
  // AsyncValidatorFn,
  Errors,
  GroupChildControls,
  ListChildControls,
  Valid,
  GroupValue,
} from './types';

/**
 * validators
 */

export {
  /**
   * @deprecated
   * 不建议直接使用 请移步ValidatorFactory
   */
  requiredValidator,
  /**
   * @deprecated
   * 不建议直接使用 请移步ValidatorFactory
   */
  requiredTrueValidator,
  /**
   * @deprecated
   * 不建议直接使用 请移步ValidatorFactory
   */
  minValidator,
  /**
   * @deprecated
   * 不建议直接使用 请移步ValidatorFactory
   */
  maxValidator,
  /**
   * @deprecated
   * 不建议直接使用 请移步ValidatorFactory
   */
  emailValidator,
  /**
   * @deprecated
   * 不建议直接使用 请移步ValidatorFactory
   */
  minLengthValidator,
  /**
   * @deprecated
   * 不建议直接使用 请移步ValidatorFactory
   */
  maxLengthValidator,
  /**
   * @deprecated
   * 不建议直接使用 请移步ValidatorFactory
   */
  nullValidator,
  /**
   * @deprecated
   * 不建议直接使用 请移步ValidatorFactory
   */
  patternValidator,
} from './validators';

export {
  requiredValidatorFactory,
  requiredTrueValidatorFactory,
  minValidatorFactory,
  maxValidatorFactory,
  emailValidatorFactory,
  minLengthValidatorFactory,
  maxLengthValidatorFactory,
  patternValidatorFactory,
} from './validators/validatorMessageFactories';

export { deepCheckFirstInvalidControl } from './utils/checkControlUtils';
