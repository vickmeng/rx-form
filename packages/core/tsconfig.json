{
  "compileOnSave": true,
  "compilerOptions": {
    "target": "esnext",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "outDir": "dist",
    "allowJs": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "strict": false,
    "forceConsistentCasingInFileNames": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "noEmit": true,
    "declaration": true,
  },
  "include": [
    "src/**/*",
  ],
  "exclude": [
    "node_modules",
    "lib",
    "es",
    "dist",
    "typings",
    "**/__test__",
    "test",
    "docs",
    "tests"
  ]
}
