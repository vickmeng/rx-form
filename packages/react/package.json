{
  "name": "@yzw-rx-form/react",
  "version": "1.2.0",
  "description": "core",
  "module": "dist/index.js",
  "types": "dist/index.d.ts",
  "main": "dist/index.js",
  "private": false,
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "rollup -c",
    "watch": "rollup -c --watch"
  },
  "author": "laomeng",
  "license": "ISC",
  "peerDependencies": {
    "react": "*",
    "@yzw-rx-form/core": "workspace:*"
  },
  "devDependencies": {
    "react": "*",
    "@yzw-rx-form/core": "workspace:*"
  }
}
