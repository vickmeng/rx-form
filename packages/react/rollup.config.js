import typescript from '@rollup/plugin-typescript';

export default {
  input: 'src/index.ts',
  output: {
    file: 'dist/index.js',
    format: 'es',
  },
  external: ['react'],
  plugins: [
    typescript({
      tsconfig: './tsconfig.json',
    }),
  ],
};
