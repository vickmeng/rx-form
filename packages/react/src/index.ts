/**
 * components
 */
export { Field } from './items/field';
export { Group } from './items/group';
export { List } from './items/list';
export { Error } from './items/error';

export type { FieldInternalProps, GroupInternalProps, ListInternalProps, ErrorInternalProps } from './types';

export {
  useControlValue,
  useControlDisabled,
  useControlDirty,
  useControlValid,
  useControlErrors,
  useControlControls,
  useControlAsyncErrors,
} from './hooks';
