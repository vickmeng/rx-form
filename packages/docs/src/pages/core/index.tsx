import React from "react";
import { Typography } from "@material-ui/core";
import { Link } from "react-router-dom";

const Core = () => {
  return (
    <div className="page">
      <Typography paragraph variant={"h2"}>
        核心概念
      </Typography>

      <Typography paragraph>react-observable-form主要包含两个概念：Controller与Component。</Typography>

      <Typography paragraph variant={"h4"}>
        Controllers
      </Typography>

      <Typography paragraph>
        Controller是数据的源头，它为视图提供一系列可观察对象，以及操作数据的方法，使用者可以通过订阅这些可观察对象获得如：value，errors等状态。
      </Typography>

      <Typography paragraph>我们有如下三种class创建Controller：</Typography>

      <Typography paragraph>
        <ul>
          <li>FieldControl：用以创建单一元素的控制器</li>
          <li>GroupControl：用以创建群组元素的控制器</li>
          <li>ListControl：用以创建列表元素的控制器</li>
        </ul>
      </Typography>

      <Typography paragraph variant={"h4"}>
        Components
      </Typography>

      <Typography paragraph>
        就是React组件，作为Controller的订阅者，可以获取其订阅的Controller的值向下传递，也提供方法让下级组件将状态上传，从而实现受控组件。
      </Typography>

      <Typography paragraph>我们有如下三种Components：</Typography>

      <Typography paragraph>
        <ul>
          <li> {"<Field/>"}：FieldControl的订阅器</li>
          <li> {"<Group/>"}：GroupControl的订阅器</li>
          <li> {"<List/>"}：ListControl的订阅器</li>
        </ul>
      </Typography>

      <br />

      <Typography paragraph>
        可以查看<Link to={"/field"}>基础使用</Link>更直观得进行学习
      </Typography>
    </div>
  );
};

export default Core;
