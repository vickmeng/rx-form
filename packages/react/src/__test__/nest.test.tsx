import React from 'react';
import { render } from '@testing-library/react';
import { GroupControl, ListControl, requiredValidator } from '@rx-form/core';

import { Field, Group, Error, List } from '../index';

describe('nest components', () => {
  it('should render correctly', () => {
    const groupControl = new GroupControl({
      name: ['', { validators: [requiredValidator] }],
      familyMembers: new ListControl([
        new GroupControl({
          name: ['Tom'],
          tel: ['13133333333'],
        }),
      ]),
    });

    const { asFragment } = render(
      <Group control={groupControl}>
        {(props) => (
          <>
            <Field<string> name="name">{({ value }) => <input value={value} readOnly />}</Field>
            <Error name="name">{({ errors }) => <>{errors?.required && '至少填一名家庭成员'}</>}</Error>

            <List name="familyMembers">
              {({ childControls }) => {
                return (
                  <>
                    {childControls.map((control, i) => {
                      return (
                        <Group name={`${i}`} key={`${i}`}>
                          {() => {
                            return (
                              <>
                                <Field<string> name="name">{({ value }) => <input value={value} readOnly />}</Field>
                                <Field<string> name="tel">{({ value }) => <input value={value} readOnly />}</Field>
                              </>
                            );
                          }}
                        </Group>
                      );
                    })}
                  </>
                );
              }}
            </List>
          </>
        )}
      </Group>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
