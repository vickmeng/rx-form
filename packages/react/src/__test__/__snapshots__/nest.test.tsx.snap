// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nest components should render correctly 1`] = `
<DocumentFragment>
  <input
    readonly=""
    value=""
  />
  至少填一名家庭成员
  <input
    readonly=""
    value="Tom"
  />
  <input
    readonly=""
    value="13133333333"
  />
</DocumentFragment>
`;
